internal/zinc-core/src/main/scala/sbt/internal/inc/Lookup.scala

```diff
@@ -76,7 +76,10 @@ trait ExternalLookup extends ExternalHooks.Lookup {
    *         Optional.of(analyzed) if analyzed.provenance.isEmpty => Couldn't find it.
    *         Optional.of(analyzed) => good
    */
-  def lookupAnalyzedClass(binaryClassName: String): Option[AnalyzedClass]
+  def lookupAnalyzedClass(
+      file: Option[VirtualFileRef],
+      binaryClassName: String,
+  ): Option[AnalyzedClass]
```

Comment:
I'd flip the parameter order here, since `file` is more or a hint.

Also I think it would be helpful to provide a documentation for what it intends to do when `Some(...)` is passed, for example in terms of how this relates to provenance. Speaking of which, provenance is a String should this be String also?

Reply:
> I'd flip the parameter order here, since `file` is more or a hint.

Sure.

> Also I think it would be helpful to provide a documentation for what it intends to do when `Some(...)` is passed, for example in terms of how this relates to provenance.

That's up to the build tool.  The file here is based on the "classFile" for the given binary class name:

https://github.com/sbt/zinc/blob/e226ba97388d4addbde9298a1bc4040ffa5d3390/internal/compiler-bridge/src/main/scala/xsbt/Dependency.scala#L177-L191

which may be a class file or a jar file:

https://github.com/sbt/zinc/blob/e226ba97388d4addbde9298a1bc4040ffa5d3390/internal/compiler-bridge/src/main/scala/xsbt/Dependency.scala#L130-L145

Provenance is a more abstract, which is why it's just a string identifier.